tests/test_other.py

```diff
@@ -2260,17 +2260,32 @@ def test_debuginfo(self):
           else:
             self.assertIn('strip-debug', opts)
 
-  def test_scons(self): # also incidentally tests c++11 integration in llvm 3.1
+  def test_scons(self):
+    # this test copies the site_scons directory alongside the test
     scons_path = Building.which('scons')
     if not scons_path:
       self.skipTest('Skipping other.test_scons: The tool "scons" was not found in PATH!')
-    try_delete(os.path.join(self.get_dir(), 'test'))
-    shutil.copytree(path_from_root('tests', 'scons'), os.path.join(self.get_dir(), 'test'))
-    shutil.copytree(path_from_root('tools', 'scons', 'site_scons'), os.path.join(self.get_dir(), 'test', 'site_scons'))
-    os.chdir(os.path.join(self.get_dir(), 'test'))
-    run_process(['scons'])
-    output = run_js('scons_integration.js', assert_returncode=5)
-    assert 'If you see this - the world is all right!' in output
+    shutil.copytree(path_from_root('tests', 'scons'), 'test')
+    shutil.copytree(path_from_root('tools', 'scons', 'site_scons'), os.path.join('test', 'site_scons'))
+    with chdir('test'):
+      with env_modify({'EMSCRIPTEN_ROOT': path_from_root()}):
+        run_process(['scons'])
+      output = run_js('scons_integration.js', assert_returncode=5)
+    self.assertContained('If you see this - the world is all right!', output)
```

Comment:
Nit: Feels weird to me to have an assert on the output be at a different level of indentation than the run command that produced it. On the other hand, this minimizes the scope of the `with`s, and less context is good. Either way.